fms/utils/fusion.py

```diff
@@ -0,0 +1,41 @@
+import torch
+import torch.nn as nn
+
+
+def _fusion_converted(module: nn.Module, fuse: bool):
+    if not fuse and hasattr(module, "unfuse"):
+        result = module.unfuse()
+        del module
+    elif fuse and hasattr(module, "fuse"):
+        result = module.fuse()
+        del module
+    else:
+        result = module
+    return result
+
+
+def apply_fusion(module: nn.Module, fuse: bool) -> nn.Module:
```

Comment:
Since normally this will only be used to unfuse, the name/signature and lack of a default seems a bit unintuitive.
Maybe to make it very clear we should just have two separate functions `fuse_kqv` and `unfuse_kqv` ?

Reply:
This is not only qkv, this will also fuse/unfuse mlp. Idea was it will look for any modules that are fuse/unfusable

Reply:
We could replace this with 2 calls like:

```
fused_model = fuse_module_weights(model)

unfused_model = unfuse_module_weights(model)
```